src/util/side_metadata/global.rs

```diff
@@ -72,129 +77,252 @@ impl MappingState {
         }
-        aligned_start += BYTES_IN_CHUNK;
+    }
+
+    if cfg!(target_pointer_width = "32") {
+        return try_map_per_chunk_metadata_space(start, size, lsize);
     }
 
     true
 }
 
-// Try to map side metadata for the chunk starting at `start`
-fn try_mmap_metadata_chunk(
-    start: Address,
-    global_per_chunk: usize,
-    local_per_chunk: usize,
-) -> MappingState {
-    trace!(
-        "try_mmap_metadata_chunk({}, 0x{:x}, 0x{:x})",
-        start,
-        global_per_chunk,
-        local_per_chunk
-    );
-    let global_meta_start = address_to_meta_chunk_addr(start);
+// Try to map side metadata for the data starting at `start` and a size of `size`
+fn try_mmap_metadata(start: Address, size: usize) -> MappingState {
+    trace!("try_mmap_metadata({}, 0x{:x})", start, size);
 
-    let prot = libc::PROT_READ | libc::PROT_WRITE | libc::PROT_EXEC;
+    debug_assert!(size > 0 && size % BYTES_IN_PAGE == 0);
+
+    let prot = libc::PROT_READ | libc::PROT_WRITE;
     // MAP_FIXED_NOREPLACE returns EEXIST if already mapped
     let flags = libc::MAP_ANON | libc::MAP_PRIVATE | libc::MAP_FIXED_NOREPLACE;
 
-    if global_per_chunk != 0 {
-        let result: *mut libc::c_void = unsafe {
-            libc::mmap(
-                global_meta_start.to_mut_ptr(),
-                global_per_chunk,
-                prot,
-                flags,
-                -1,
-                0,
-            )
-        };
-
-        if result == libc::MAP_FAILED {
-            let err = unsafe { *libc::__errno_location() };
-            if err == libc::EEXIST {
-                return MappingState::WasMapped;
-            } else {
-                return MappingState::NotMappable;
-            }
+    let result: *mut libc::c_void =
+        unsafe { libc::mmap(start.to_mut_ptr(), size, prot, flags, -1, 0) };
+
+    if result == libc::MAP_FAILED {
+        let err = unsafe { *libc::__errno_location() };
+        if err == libc::EEXIST {
+            debug!("try_mmap_metadata({}, 0x{:x}) -> WasMapped", start, size);
+            return MappingState::WasMapped;
+        } else {
+            debug!("try_mmap_metadata({}, 0x{:x}) -> NotMappable", start, size);
+            return MappingState::NotMappable;
         }
     }
 
-    let policy_meta_start = global_meta_start + POLICY_SIDE_METADATA_OFFSET;
-    if local_per_chunk != 0 {
-        let result: *mut libc::c_void = unsafe {
-            libc::mmap(
-                policy_meta_start.to_mut_ptr(),
-                local_per_chunk,
-                prot,
-                flags,
-                -1,
-                0,
-            )
-        };
-
-        if result == libc::MAP_FAILED {
-            let err = unsafe { *libc::__errno_location() };
-            if err == libc::EEXIST {
-                return MappingState::WasMapped;
+    MappingState::IsMapped
+}
+
+fn try_mmap_metadata_address_range(start: Address, size: usize) -> bool {
```

Comment:
What is the difference between `mmap_metadata_address_range` and `mmap_metadata`? They take the same arguments, but returns different types. They use different flags for mmap. They have different checks after mmapping. This is very confusing.

Reply:
Do you think changing `try_mmap_metadata_address_range` to `try_mmap_noreserve_metadata` makes it more clear?
This function only maps addresses, but doesn't reserve any swap-space/physical memory.

Reply:
or `try_map_metadata_address_range` to `try_reserve_metadata_addresses`

Reply:
I suggest making it clear what those functions do, why we need 2 functions that are similar, in what cases we should use one function rather than the other. You can achieve this by adding more comments, renaming the functions, or refactored the code so the code is self explanatory.